app/services/rollover_service.rb

```diff
@@ -0,0 +1,61 @@
+class RolloverService
+  def initialize(args)
+    @args = args
+  end
+
+  class << self
+    def call(args)
+      new(args).call
+    end
+  end
+
+
+  def call
+    providers = if @args.any?
+                  RecruitmentCycle.current_recruitment_cycle
+                                  .providers.where(provider_code: @args.to_a.map(&:upcase))
+                else
+                  RecruitmentCycle.current_recruitment_cycle.providers
+                end
+
+    new_recruitment_cycle = RecruitmentCycle.next_recruitment_cycle
+
+    total_counts = { providers: 0, sites: 0, courses: 0 }
+
+    total_bm = Benchmark.measure do
+      providers.each do |provider|
+        print "Copying provider #{provider.provider_code}: "
+        counts = nil
+        bm = Benchmark.measure do
+          Provider.connection.transaction do
+            copy_courses_to_provider_service = Courses::CopyToProviderService.new(
+              sites_copy_to_course: Sites::CopyToCourseService.new,
+              enrichments_copy_to_course: Enrichments::CopyToCourseService.new,
+              )
+
+            copy_provider_to_recruitment_cycle = Providers::CopyToRecruitmentCycleService.new(
+              copy_course_to_provider_service: copy_courses_to_provider_service,
+              copy_site_to_provider_service: Sites::CopyToProviderService.new,
+              logger: Logger.new(STDOUT),
+              )
+
+            counts = copy_provider_to_recruitment_cycle.execute(
+              provider: provider, new_recruitment_cycle: new_recruitment_cycle,
+              )
+          end
+        end
+        puts "provider #{counts[:providers] ? 'copied' : 'skipped'}, " \
```

Comment:
I don't think this ever prints `skipped`, even if we are skipping the provider. Based on the code in `Providers::CopyToRecruitmentCycleService` I think it should be: 
```suggestion
        puts "provider #{counts[:providers].zero? ? 'skipped' : 'copied'}, " \
```